_posts/2025-06-24-kafka-queue.md

```diff
@@ -0,0 +1,703 @@
+---
+layout: post
+title:  "Using Queues for Kafka with Strimzi"
+date: 2025-06-24
+author: tina_selenge
+---
+
+The Queues for Kafka feature was introduced by [KIP-932](https://cwiki.apache.org/confluence/display/KAFKA/KIP-932%3A+Queues+for+Kafka#KIP932:QueuesforKafka) and is in early access in Apache Kafka version 4.0 which is supported by Strimzi 0.46. In this blog, I’m going to introduce this feature and show how you can try it out with your Strimzi managed cluster. 
+
+This feature is only supported with Kafka clusters running in KRaft mode, since Zookeeper was removed in the Apache Kafka 4.0 release. It is also based on the new consumer rebalance protocol introduced by [KIP-848](https://cwiki.apache.org/confluence/display/KAFKA/KIP-848%3A+The+Next+Generation+of+the+Consumer+Rebalance+Protocol) that enhances stability, scalability and performance. 
+
+Traditional message queues and Kafka topics are significantly different in terms of design and use cases. A queue allows a group of consumers to read and process records in parallel to distribute the workload. It does not typically provide a strong ordering guarantee as records in a queue are consumed and processed independently. However it does provide exactly-once guarantee as each record is only read once and is deleted from the queue. States of individual records are usually tracked and acknowledged after being processed successfully to avoid prematurely removing records off the queue or to handle unprocessable records. This also means that records in a queue are not replayable, therefore a queue can only be consumed by a single group of consumers.
```

Comment:
> However it does provide exactly-once guarantee as each record is only read once and is deleted from the queue.

I would not say that traditional queueing system needs to have exactly once guarantees. At least once guarantees are pretty common. Given at the end of this section you say that Kafka does not support exactly-once for queues, maybe you can be less strict about it here? 😉